`MjCompile` intermittently throws `NullPointerException` during concurrent `eo-runtime` compilation due to race condition

I cannot reliably build `eo-runtime` because `MjCompile` intermittently fails with a `NullPointerException` inside the xsline parsing pipeline. Running the same command locally always succeeds, and restarting the CI pipeline eventually heals it, which strongly points to a race condition.

**Steps to reproduce**

Run the qulice CI pipeline on any branch that touches `eo-runtime`. The failure is non-deterministic — it surfaces occasionally under load (CI parallelism) and disappears on retry.

**Expected behaviour**

`MjCompile` should succeed consistently regardless of thread interleaving.

**Actual behaviour**

```
Failed to process "tt.sscanf" (org.eolang.maven.TjForeign):
Cannot invoke "String.getBytes(java.nio.charset.Charset)" because "xml" is null
```

The null `xml` string reaches `DomParser.BytesSource` via `StXSL.lambda$new$0`, meaning one thread in the `Threaded` pool receives a null XML string while the xsline transformation pipeline is processing the EO source.

The likely culprit is `ConcurrentCache`: each call site creates a **new** `ConcurrentCache` instance with its own `ConcurrentHashMap` of locks, so two concurrent invocations that happen to target the same cache path will each hold *different* locks and can race on the same file — one thread reading a partially-written or empty file while another thread is still writing it.

<details>
<summary>Full stack trace</summary>

```
Caused by: java.lang.NullPointerException: Cannot invoke "String.getBytes(java.nio.charset.Charset)" because "xml" is null
    at com.jcabi.xml.DomParser$BytesSource.<init> (DomParser.java:209)
    at com.jcabi.xml.DomParser.<init> (DomParser.java:55)
    at com.jcabi.xml.XMLDocument.<init> (XMLDocument.java:122)
    at com.yegor256.xsline.StXSL.lambda$new$0 (StXSL.java:44)
    at com.yegor256.xsline.StLambda.uid (StLambda.java:108)
    at com.yegor256.xsline.StEnvelope.uid (StEnvelope.java:56)
    at com.yegor256.xsline.StLogged.uid (StLogged.java:100)
    at com.yegor256.xsline.StFast.uid (StFast.java:88)
    at com.yegor256.xsline.StFast.apply (StFast.java:100)
    at com.yegor256.xsline.StAfter.lambda$new$0 (StAfter.java:48)
    at com.yegor256.xsline.StLambda.apply (StLambda.java:115)
    at com.yegor256.xsline.Xsline.pass (Xsline.java:85)
    at org.eolang.parser.EoSyntax.parsed (EoSyntax.java:121)
    at org.eolang.maven.EoSource.parsed (EoSource.java:65)
    at org.eolang.maven.Parsing.parsed (Parsing.java:205)
    at org.eolang.maven.Parsing.lambda$parsed$0 (Parsing.java:164)
    at org.eolang.maven.Cache.apply (Cache.java:93)
    at org.eolang.maven.ConcurrentCache.apply (ConcurrentCache.java:59)
    at org.eolang.maven.Parsing.parsed (Parsing.java:169)
    at org.eolang.maven.Threaded.lambda$total$1 (Threaded.java:76)
    at java.util.concurrent.FutureTask.run (FutureTask.java:317)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1144)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:642)
    at java.lang.Thread.run (Thread.java:1583)
```
</details>

Relevant classes to investigate: `ConcurrentCache`, `Cache`, `Parsing`, `Threaded`.

```
Error:  org.eolang.maven.MjParseTest.addsErrorsWhenObjectNameFails(Path) -- Time elapsed: 1.476 s <<< ERROR!
java.lang.IllegalStateException: org.apache.maven.plugin.MojoFailureException: 'MjParse' execution failed
	at org.eolang.maven.Moja.execute(Moja.java:112)
	at org.eolang.maven.FakeMaven.execute(FakeMaven.java:206)
	at org.eolang.maven.FakeMaven.execute(FakeMaven.java:140)
	at org.eolang.maven.MjParseTest.addsErrorsWhenObjectNameFails(MjParseTest.java:234)
Caused by: org.apache.maven.plugin.MojoFailureException: 'MjParse' execution failed
	at org.eolang.maven.MjSafe.exitError(MjSafe.java:704)
	at org.eolang.maven.MjSafe.execute(MjSafe.java:513)
	at org.eolang.maven.Moja.execute(Moja.java:104)
	... 3 more
Caused by: java.util.concurrent.ExecutionException: java.io.UncheckedIOException: java.io.IOException: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Failed to process "foo.x.main" (org.eolang.maven.TjForeign)
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:205)
	at org.eolang.maven.MjSafe.execWithTimeout(MjSafe.java:653)
	at org.eolang.maven.MjSafe.execute(MjSafe.java:495)
	... 4 more
Caused by: java.io.UncheckedIOException: java.io.IOException: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Failed to process "foo.x.main" (org.eolang.maven.TjForeign)
	at org.cactoos.scalar.Unchecked.value(Unchecked.java:39)
	at org.cactoos.iterator.Mapped$1.next(Mapped.java:49)
	at org.cactoos.iterator.IteratorEnvelope.next(IteratorEnvelope.java:40)
	at org.cactoos.scalar.Folded.value(Folded.java:62)
	at org.cactoos.number.NumberOfScalars.lambda$new$1(NumberOfScalars.java:49)
	at org.cactoos.scalar.Checked.value(Checked.java:50)
	at org.cactoos.scalar.IoChecked.value(IoChecked.java:44)
	at org.cactoos.scalar.Unchecked.value(Unchecked.java:37)
	at org.cactoos.number.NumberOfScalars.intValue(NumberOfScalars.java:74)
	at org.cactoos.number.NumberEnvelope.intValue(NumberEnvelope.java:36)
	at org.eolang.maven.Threaded.total(Threaded.java:90)
	at org.eolang.maven.Parsing.exec(Parsing.java:121)
	at org.eolang.maven.MjParse.exec(MjParse.java:45)
	at org.eolang.maven.MjSafe.lambda$execWithTimeout$8(MjSafe.java:650)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1575)
Caused by: java.io.IOException: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Failed to process "foo.x.main" (org.eolang.maven.TjForeign)
	at org.cactoos.func.UncheckedFunc.lambda$apply$0(UncheckedFunc.java:37)
	at org.cactoos.scalar.Checked.value(Checked.java:50)
	at org.cactoos.scalar.IoChecked.value(IoChecked.java:44)
	at org.cactoos.scalar.Unchecked.value(Unchecked.java:37)
	at org.cactoos.func.UncheckedFunc.apply(UncheckedFunc.java:38)
	at org.cactoos.scalar.Checked.wrappedException(Checked.java:73)
	at org.cactoos.scalar.Checked.value(Checked.java:59)
	at org.cactoos.scalar.IoChecked.value(IoChecked.java:44)
	at org.cactoos.scalar.Unchecked.value(Unchecked.java:37)
	... 17 more
Caused by: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Failed to process "foo.x.main" (org.eolang.maven.TjForeign)
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at org.cactoos.scalar.Mapped.lambda$new$0(Mapped.java:40)
	at org.cactoos.scalar.ScalarEnvelope.value(ScalarEnvelope.java:34)
	at org.cactoos.scalar.Checked.value(Checked.java:50)
	... 19 more
Caused by: java.lang.IllegalStateException: Failed to process "foo.x.main" (org.eolang.maven.TjForeign)
	at org.eolang.maven.Threaded.lambda$total$1(Threaded.java:84)
	at org.cactoos.scalar.CallableEnvelope.call(CallableEnvelope.java:34)
	... 4 more
Caused by: java.lang.NullPointerException: Cannot invoke "String.getBytes(java.nio.charset.Charset)" because "xml" is null
	at com.jcabi.xml.DomParser$BytesSource.<init>(DomParser.java:209)
	at com.jcabi.xml.DomParser.<init>(DomParser.java:55)
	at com.jcabi.xml.XMLDocument.<init>(XMLDocument.java:122)
	at com.yegor256.xsline.StXSL.lambda$new$0(StXSL.java:44)
	at com.yegor256.xsline.StLambda.uid(StLambda.java:108)
	at com.yegor256.xsline.StEnvelope.uid(StEnvelope.java:56)
	at com.yegor256.xsline.StEnvelope.uid(StEnvelope.java:56)
	at com.yegor256.xsline.StLogged.uid(StLogged.java:100)
	at com.yegor256.xsline.StFast.uid(StFast.java:88)
	at org.eolang.parser.TrStepped.lambda$new$1(TrStepped.java:64)
	at com.yegor256.xsline.StLambda.apply(StLambda.java:115)
	at com.yegor256.xsline.StAfter.lambda$new$0(StAfter.java:50)
	at com.yegor256.xsline.StLambda.apply(StLambda.java:115)
	at com.yegor256.xsline.StEnvelope.apply(StEnvelope.java:61)
	at com.yegor256.xsline.Xsline.pass(Xsline.java:85)
	at org.eolang.parser.EoSyntax.parsed(EoSyntax.java:121)
	at org.eolang.maven.EoSource.parsed(EoSource.java:65)
	at org.eolang.maven.Parsing.parsed(Parsing.java:205)
	at org.eolang.maven.Parsing.lambda$parsed$0(Parsing.java:164)
	at org.cactoos.func.UncheckedFunc.lambda$apply$0(UncheckedFunc.java:37)
	at org.cactoos.scalar.Checked.value(Checked.java:50)
	at org.cactoos.scalar.IoChecked.value(IoChecked.java:44)
	at org.cactoos.scalar.Unchecked.value(Unchecked.java:37)
	at org.cactoos.func.UncheckedFunc.apply(UncheckedFunc.java:38)
	at org.eolang.maven.Cache.apply(Cache.java:93)
	at org.eolang.maven.ConcurrentCache.apply(ConcurrentCache.java:59)
	at org.eolang.maven.Parsing.parsed(Parsing.java:169)
	at org.eolang.maven.Threaded.lambda$total$1(Threaded.java:76)
	... 5 more
```

--- a/pom.xml
+++ b/pom.xml
@@ -205,7 +205,7 @@
       <dependency>
         <groupId>com.jcabi</groupId>
         <artifactId>jcabi-xml</artifactId>
-        <version>0.37.0</version>
+        <version>0.37.1</version>
       </dependency>
       <dependency>
         <groupId>com.jcabi</groupId>

